File: ch200203/settlement-api/src/main/java/com/hanwha/settlement/users/UserService.java
package com.hanwha.settlement.users;

import com.hanwha.settlement.users.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    @Transactional(readOnly = true)
    public User findUser(final long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 유저 입니다 : " + id));
    }

}

File: mjkim103301/pay-separately/src/main/kotlin/org/example/pay/service/UserService.kt
package org.example.pay.service

import org.example.pay.repository.UserRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class UserService @Autowired constructor(private val userRepository: UserRepository) {
    fun findUser(userId: Long) = userRepository.findById(userId)
}
